src/sys/socket/sockopt.rs

```diff
@@ -1424,4 +1551,71 @@ mod test {
         assert!(s_listening2);
         close(s).unwrap();
     }
+
+    #[cfg(any(target_os = "android", target_os = "linux"))]
+    #[test]
+    fn netlink_list_memberships() {
+        use super::super::*;
+        use crate::unistd::close;
+
+        let s = socket(
+            AddressFamily::Netlink,
+            SockType::Raw,
+            SockFlag::empty(),
+            SockProtocol::NetlinkRoute,
+        );
+
+        if let Err(Errno::EPROTONOSUPPORT) = s {
+            return;
+        }
+
+        let s = s.unwrap();
+
+        let groups = || -> Result<Vec<u32>> {
```

Comment:
Why define a closure and then immediately call it?  That's confusing.

Reply:
ACK